slsReceiverSoftware/src/slsReceiverTCPIPInterface.cpp

```diff
@@ -682,1580 +547,793 @@ int slsReceiverTCPIPInterface::set_roi(sls::ServerInterface2 &socket) {
+	strcpy(retval, s.c_str());
+	if ((!s.length()) || (strlen(fPath) && strcasecmp(fPath, retval)))
+		throw RuntimeError("Receiver file path does not exist");
+	else
+		FILE_LOG(logDEBUG1) << "file path:" << retval;
+	
+    return socket.sendResult(retval);
+}
 
-	if (ret == OK) {
-		FILE_LOG(logDEBUG1) << "Getting Status";
-		retval = receiver->getStatus();
-		FILE_LOG(logDEBUG1) << "Status:" << runStatusType(retval);
+int slsReceiverTCPIPInterface::set_file_name(Interface &socket) {
+	char fName[MAX_STR_LENGTH]{};
+    char retval[MAX_STR_LENGTH]{};
+	socket.receiveArg(fName);
+	if (strlen(fName)) {
+		FILE_LOG(logDEBUG1) << "Setting file name: " << fName;
+		impl()->setFileName(fName);
 	}
-	return socket.sendResult(true, ret, &retval, sizeof(retval), mess);
+	std::string s = impl()->getFileName();
+	if (s.empty())
+		throw RuntimeError("file name is empty");
+
+	strcpy(retval, s.c_str());
+	FILE_LOG(logDEBUG1) << "file name:" << retval;
+	return socket.sendResult(retval);
 }
 
+int slsReceiverTCPIPInterface::set_file_index(Interface &socket) {
+	auto index = socket.receive<int>();
+	if (index >= 0) {
+		VerifyIdle(socket);
+		FILE_LOG(logDEBUG1) << "Setting file index: " << index;
+		impl()->setFileIndex(index);
+	}
+	int retval = impl()->getFileIndex();
+	validate(index, retval, std::string("set file index"), DEC);
+	FILE_LOG(logDEBUG1) << "file index:" << retval;
+	return socket.sendResult(retval);
+}
 
+int	slsReceiverTCPIPInterface::get_frame_index(Interface &socket){
+	uint64_t retval = impl()->getAcquisitionIndex();
+	FILE_LOG(logDEBUG1) << "frame index:" << retval;
+	return socket.sendResult(retval);
+}
 
-int slsReceiverTCPIPInterface::start_receiver(sls::ServerInterface2 &socket){
-	ret = OK;
-	memset(mess, 0, sizeof(mess));
+int	slsReceiverTCPIPInterface::get_frames_caught(Interface &socket){
+	int retval = impl()->getTotalFramesCaught();
+	FILE_LOG(logDEBUG1) << "frames caught:" << retval;
+	return socket.sendResult(retval);
+}
 
-	if(receiver == nullptr){
-		NullObjectError(ret, mess);
-	}
-	// receiver is not null
-	if (ret == OK) {
-		// only set
-		// verify if receiver is unlocked
-		if (VerifyLock(ret, mess) == OK) {
-			// should not be idle
-			enum runStatus s = receiver->getStatus();
-			if (s != IDLE) {
-				ret=FAIL;
-				sprintf(mess,"Cannot start Receiver as it is in %s state\n",runStatusType(s).c_str());
-				FILE_LOG(logERROR) << mess;
-			}else {
-				FILE_LOG(logDEBUG1) << "Starting Receiver";
-				ret = receiver->startReceiver(mess);
-				if (ret == FAIL) {
-					FILE_LOG(logERROR) << mess;
-				}
-			}
-		}
+int	slsReceiverTCPIPInterface::reset_frames_caught(Interface &socket){
+	FILE_LOG(logDEBUG1) << "Reset frames caught";
+	impl()->resetAcquisitionCount();
+	return socket.sendData(OK);
+}
+
+int slsReceiverTCPIPInterface::enable_file_write(Interface &socket){
+	auto enable = socket.receive<int>();
+	// set
+	if (enable >= 0) {
+		// verify if receiver is unlocked and idle
```

Comment:
minor. you can remove this comment as it does not check unlocked here

Reply:
done!